File: astrid/src/main/java/com/todoroo/astrid/helper/UUIDHelper.java
package com.todoroo.astrid.helper;

import java.util.UUID;

public class UUIDHelper {

    private static final long MIN_UUID = 100000000;

    /**
     * @return a pair consisting of the newly
     * generated uuid and the corresponding proof text
     */
    public static String newUUID() {
        long uuid = 0;
        do {
            uuid = UUID.randomUUID().getLeastSignificantBits() & 0x7fffffffffffffffL;;
        } while (uuid < MIN_UUID);
        return Long.toString(uuid);
    }
}

File: data/src/main/kotlin/org/tasks/data/UUIDHelper.kt
package org.tasks.data

import java.util.UUID

object UUIDHelper {
    private const val MIN_UUID: Long = 100000000

    fun newUUID(): String {
        var uuid: Long
        do {
            uuid = UUID.randomUUID().leastSignificantBits and 0x7fffffffffffffffL
        } while (uuid < MIN_UUID)
        return uuid.toString()
    }
}
